### PR TITLE
initial open sourcing commit

### PR DESCRIPTION
**Pull request by hadasah**
Opening a draft PR for comments

### CODE REVIEW

File: projects/unlikelihood/agents.py [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+import torch.nn.functional as F
+import json
+from collections import defaultdict, Counter
+from nltk import ngrams
+
+from parlai.utils.misc import round_sigfigs
+from parlai.agents.transformer.transformer import TransformerGeneratorAgent
+from parlai.core.torch_agent import TorchAgent, Batch, Output
+
+from parlai_internal.agents.parlall.agents import ParlallAgent
```
Comment (hadasah) [inline projects/unlikelihood/agents.py]: @stephenroller  I don't think parlall is open sourced yet

File: projects/unlikelihood/agents.py [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+
+    def receive_metrics(self, metrics_dict):
+        super().receive_metrics(metrics_dict)
+        if self.opt['train_to_convergence']:
+            return
+        if False:
+            if self.opt['task'] == 'convai2':
+                if metrics_dict['ppl'] > 12 or metrics_dict['f1'] < 0.19:
+                    raise StopTrainException('Convai2 PPL is too high or f1 is too low')
+            elif self.opt['task'] == 'internal:eli5':
```
Comment (hadasah) [inline projects/unlikelihood/agents.py]: same for eli5

File: projects/unlikelihood/agents.py (hunk 1 above)
Comment (stephenroller) [reply projects/unlikelihood/agents.py]: It is, it goes by the name image_seq2seq. We can strip it for now I think.
